Sources/AirbnbSwiftFormatTool/AirbnbSwiftFormatTool.swift

```diff
@@ -40,40 +40,74 @@ struct AirbnbSwiftFormatTool: ParsableCommand {
 
-  mutating func run() throws {
-    try swiftFormat.run()
-    swiftFormat.waitUntilExit()
-
-    try swiftLint.run()
-    swiftLint.waitUntilExit()
-
-    if log {
-      log(swiftFormat.shellCommand)
-      log(swiftLint.shellCommand)
-      log("SwiftFormat ended with exit code \(swiftFormat.terminationStatus)")
-      log("SwiftLint ended with exit code \(swiftLint.terminationStatus)")
+  func run() throws {
+    let swiftFormat = makeSwiftFormatCommand()
+    let swiftLint = makeSwiftLintCommand(autocorrect: false)
+    let swiftLintAutocorrect = makeSwiftLintCommand(autocorrect: true)
+
+    let swiftFormatExitCode = try swiftFormat.run()
+
+    // We always have to run SwiftLint in lint-only mode at least once,
+    // because when in autocorrect mode SwiftLint won't emit any lint warnings.
+    let swiftLintExitCode = try swiftLint.run()
+
+    let swiftLintAutocorrectExitCode: Int32?
+    if
+      // When only linting, we shouldn't run SwiftLint with autocorrect enabled
+      !lintOnly,
+      // When formatting, only re-run SwiftLint a second time if the first
+      // invocation reported a lint failure that needs to be corrected
+      swiftLintExitCode == SwiftLintExitCode.lintFailure
+    {
+      swiftLintAutocorrectExitCode = try swiftLintAutocorrect.run()
+    } else {
+      swiftLintAutocorrectExitCode = nil
     }
 
     if
-      swiftFormat.terminationStatus == SwiftFormatExitCode.lintFailure ||
-      swiftLint.terminationStatus == SwiftLintExitCode.lintFailure
+      swiftFormatExitCode == SwiftFormatExitCode.lintFailure ||
+      swiftLintExitCode == SwiftLintExitCode.lintFailure ||
+      swiftLintAutocorrectExitCode == SwiftLintExitCode.lintFailure
     {
       throw ExitCode.failure
     }
 
     // Any other non-success exit code is an unknown failure
-    if swiftFormat.terminationStatus != EXIT_SUCCESS {
-      throw ExitCode(swiftFormat.terminationStatus)
+    if swiftFormatExitCode != EXIT_SUCCESS {
+      throw ExitCode(swiftFormatExitCode)
     }
 
-    if swiftLint.terminationStatus != EXIT_SUCCESS {
-      throw ExitCode(swiftLint.terminationStatus)
+    if swiftLintExitCode != EXIT_SUCCESS {
+      throw ExitCode(swiftLintExitCode)
+    }
+
+    if
+      let swiftLintAutocorrectExitCode = swiftLintAutocorrectExitCode,
+      swiftLintAutocorrectExitCode != EXIT_SUCCESS
+    {
+      throw ExitCode(swiftLintAutocorrectExitCode)
     }
   }
 
+  /// Runs the `AirbnbSwiftFormatTool` command with the given closure that executes each individual command.
+  func run(executeCommand: @escaping (Command) -> Int32) throws {
+    let existingImplementation = Command.runCommand
+
+    Command.runCommand = executeCommand
+    defer { Command.runCommand = existingImplementation }
+
+    try run()
+  }
+
   // MARK: Private
 
-  private lazy var swiftFormat: Process = {
+  /// Whether the command should autocorrect invalid code, or only emit lint errors
+  private var lintOnly: Bool {
+    lint
+  }
```

Comment:
What do you think about updating the `lint` property to be called `lintOnly`? This would be a breaking API change but I think it would increase the clarity of the CLI interface too. `--lint-only` seems more clear than `--lint` if the default behavior is autocorrection.

Reply:
I think `--lint` is alright. SwiftFormat uses `swiftformat --lint` to mean "only lint, don't apply any changes" as an example.

Reply:
If we were starting from scratch I'd be cool with the `--lint-only` suggestion but given we already have consumers of this tool I'd prefer to not make breaking changes

Reply:
sg